File: coffeescript/euler1_5.c
'Euler1 in CoffeeScript';
var myFilter, myMap, myReduce;

myMap = function(size) {
  var i, _results;
  _results = [];
  for (i = 1; 1 <= size ? i <= size : i >= size; 1 <= size ? i++ : i--) {
    _results.push(i);
  }
  return _results;
};

myFilter = function(ints) {
  var i, _i, _len, _results;
  _results = [];
  for (_i = 0, _len = ints.length; _i < _len; _i++) {
    i = ints[_i];
    if (i % 3 === 0 || i % 5 === 0) _results.push(i);
  }
  return _results;
};

myReduce = function(ints) {
  return ints.reduce(function(x, y) {
    return x + y;
  });
};

alert(myReduce(myFilter(myMap(999))));

File: clojure/euler1_5.clj
; Euler1 in Clojure

(defn myMap [lst]
    (map (fn [x] x) lst))

(defn myFilter [lst]
    (filter
        #(or (= (mod % 3) 0) (= (mod % 5) 0))
        lst))

(defn myReduce [lst]
    (reduce + lst))

(defn euler1 [lst]
    (myReduce (myFilter (myMap lst))))

(pr (euler1 (range 1000)))